woocommerce/woocommerce
# Wrong branch, ignore

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #41976 and includes a small refactor so that the two `.animate()` calls are now using the same keyframes format.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

To test the `.fadeIn()` replace:

1. Delete this piece of code from `plugins\woocommerce\assets\js\frontend\single-product.min.js` in order to allow a Single Product Page with a single image to initialize FlexSlider:

`1===this.$images.length&&(this.flexslider_enabled=!1),`

By default, pages with a single image don't initialize FlexSlider at all, which begs the question of whether we should straight up delete the `.fadeIn()` code in the future.

2. Clear your browser's cache so that `single-product.min.js` gets updated.
3. Navigate to a Single Product Page containing only a single image.
4. Confirm that the single image is displayed with a fadeIn animation.

To test the `.animate()` keyframes refactor:

1. Add this custom code to your theme's `functions.php`:
```
function carousel_options( $options ) {
	return array_merge( $options, array( 'useCSS' => false ) );
}
add_filter( 'woocommerce_single_product_carousel_options', 'carousel_options' )
```
2. Navigate to a Single Product Page containing multiple images.
3. Confirm that you are able to switch between the multiple images.
